python/tests/unit/src/common/utils/types/pb_wrappers/gigl_resource_config_test.py

```diff
@@ -0,0 +1,596 @@
+    GiglResourceConfigWrapper,
+)
+from snapchat.research.gbml import gigl_resource_config_pb2
+
+
+class TestGiglResourceConfigWrapper(unittest.TestCase):
+    """Test suite for GiglResourceConfigWrapper."""
+
+    def setUp(self) -> None:
+        """Set up test fixtures before each test method."""
+        self.proto_utils = ProtoUtils()
+
+    def _create_shared_resource_config(
+        self,
+    ) -> gigl_resource_config_pb2.SharedResourceConfig:
+        """Helper to create a valid SharedResourceConfig."""
+        config = gigl_resource_config_pb2.SharedResourceConfig()
+        config.resource_labels["env"] = "test"
+        config.resource_labels["cost_resource_group_tag"] = "unittest_COMPONENT"
+        config.resource_labels["cost_resource_group"] = "gigl_test"
+        config.common_compute_config.project = "test-project"
+        config.common_compute_config.region = "us-central1"
+        config.common_compute_config.temp_assets_bucket = "gs://test-temp-bucket"
+        config.common_compute_config.temp_regional_assets_bucket = (
+            "gs://test-temp-regional-bucket"
+        )
+        config.common_compute_config.perm_assets_bucket = "gs://test-perm-bucket"
+        config.common_compute_config.temp_assets_bq_dataset_name = "test_temp_dataset"
+        config.common_compute_config.embedding_bq_dataset_name = (
+            "test_embeddings_dataset"
+        )
+        config.common_compute_config.gcp_service_account_email = (
+            "test-sa@test-project.iam.gserviceaccount.com"
+        )
+        config.common_compute_config.dataflow_runner = "DataflowRunner"
+        return config
+
+    def _create_gigl_resource_config_with_direct_shared_config(
+        self,
+    ) -> gigl_resource_config_pb2.GiglResourceConfig:
+        """Helper to create a GiglResourceConfig with direct SharedResourceConfig."""
+        config = gigl_resource_config_pb2.GiglResourceConfig()
+        config.shared_resource_config.CopyFrom(self._create_shared_resource_config())
+        return config
+
+    def _create_gigl_resource_config_with_shared_config_uri(
+        self, uri: str
+    ) -> gigl_resource_config_pb2.GiglResourceConfig:
+        """Helper to create a GiglResourceConfig with SharedResourceConfig URI."""
+        config = gigl_resource_config_pb2.GiglResourceConfig()
+        config.shared_resource_config_uri = uri
+        return config
+
+    def test_shared_resource_config_direct(self):
+        """Test loading SharedResourceConfig directly."""
+        config = self._create_gigl_resource_config_with_direct_shared_config()
+        wrapper = GiglResourceConfigWrapper(resource_config=config)
+
+        shared_config = wrapper.shared_resource_config
+        self.assertEqual(shared_config.common_compute_config.project, "test-project")
+        self.assertEqual(shared_config.common_compute_config.region, "us-central1")
+        self.assertEqual(shared_config.resource_labels["env"], "test")
+
```

Comment:
w/ https://github.com/Snapchat/GiGL/pull/383#discussion_r2525459969 applied

```suggestion
        wrapper = GiglResourceConfigWrapper(resource_config=self._test_resource_config)
        self.assertEqual(wrapper.shared_resource_config, self._test_resource_config.shared_config)
```

Reply:
Done.

Creating new objects to ensure we don't mutate by accident.